### DISCUSSION TITLE
Doesn't  work on c++builder

### DISCUSSION THREAD

**Discussion by saimmm** [skia4delphi/skia4delphi#149] [Q&A] [answered]
I have RAD Alex (11),  the  library doesn't work on C++Builder. just to be honest, I am  new on c++ world. here are the errors list:
[C++ Error] Skia.Bindings.hpp(139, 25):  virtual function 'Release' has a different return type ('void') than the function it overrides (which has return type 'ULONG' (aka 'unsigned long'))
[C++ Error] unknwn.h(120, 44):  overridden virtual function is here
[C++ Error] Skia.hpp(3459, 27):  duplicate member 'ChildrenBlenders'
[C++ Error] Skia.hpp(3458, 27):  previous declaration is here
[C++ Error] Skia.hpp(3461, 31):  duplicate member 'ChildrenColorFilters'
[C++ Error] Skia.hpp(3460, 31):  previous declaration is here
[C++ Error] Skia.hpp(3464, 26):  duplicate member 'ChildrenShaders'
[C++ Error] Skia.hpp(3463, 26):  previous declaration is here
[C++ Error] Skia.hpp(3466, 38):  duplicate member 'ChildType'
[C++ Error] Skia.hpp(3465, 38):  previous declaration is here
[C++ Error] Skia.hpp(3472, 23):  duplicate member 'UniformOffset'
[C++ Error] Skia.hpp(3471, 23):  previous declaration is here
[C++ Error] Skia.hpp(3474, 19):  duplicate member 'Uniforms'
[C++ Error] Skia.hpp(3473, 19):  previous declaration is here
[C++ Error] Skia.hpp(3476, 40):  duplicate member 'UniformType'
[C++ Error] Skia.hpp(3475, 40):  previous declaration is here
[C++ Error] Skia.hpp(3478, 16):  duplicate member 'UniformTypeCount'
[C++ Error] Skia.hpp(3477, 16):  previous declaration is here
[C++ Error] Skia.hpp(5062, 113):  expected identifier
[C++ Error] Skia.Vcl.hpp(191, 58):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(191, 68):  expected class name
[C++ Error] Skia.Vcl.hpp(193, 14):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(224, 12):  no member named 'Align' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(225, 12):  no member named 'Anchors' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(226, 12):  no member named 'Constraints' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(227, 12):  no member named 'DragCursor' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(228, 12):  no member named 'DragKind' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(229, 12):  no member named 'DragMode' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(230, 12):  no member named 'Enabled' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(232, 12):  no member named 'ParentShowHint' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(233, 12):  no member named 'PopupMenu' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(234, 12):  no member named 'ShowHint' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(235, 12):  no member named 'Touch' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(236, 12):  no member named 'Visible' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(237, 12):  no member named 'OnClick' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(238, 12):  no member named 'OnContextPopup' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(239, 12):  no member named 'OnDblClick' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(240, 12):  no member named 'OnDragDrop' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(241, 12):  no member named 'OnDragOver' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(242, 12):  no member named 'OnEndDock' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(243, 12):  no member named 'OnEndDrag' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(244, 12):  no member named 'OnGesture' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(245, 12):  no member named 'OnMouseActivate' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(246, 12):  no member named 'OnMouseDown' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(247, 12):  no member named 'OnMouseEnter' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(248, 12):  no member named 'OnMouseLeave' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(249, 12):  no member named 'OnMouseMove' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(250, 12):  no member named 'OnMouseUp' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(251, 12):  no member named 'OnResize' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(252, 12):  no member named 'OnStartDock' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(253, 12):  no member named 'OnStartDrag' in 'Skia::Vcl::TSkCustomControl'
[C++ Error] Skia.Vcl.hpp(350, 61):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(350, 71):  expected class name
[C++ Error] Skia.Vcl.hpp(352, 14):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(355, 6):  'Graphics' is not a class, namespace, or enumeration
[C++ Error] Vcl.Graphics.hpp(48, 10):  'Graphics' declared here
[C++ Error] Skia.Vcl.hpp(367, 6):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(390, 12):  no member named 'Align' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(391, 12):  no member named 'Anchors' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(392, 12):  no member named 'Constraints' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(393, 12):  no member named 'DragCursor' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(394, 12):  no member named 'DragKind' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(395, 12):  no member named 'DragMode' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(396, 12):  no member named 'Enabled' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(398, 12):  no member named 'ParentShowHint' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(399, 12):  no member named 'PopupMenu' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(400, 12):  no member named 'ShowHint' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(401, 12):  no member named 'Touch' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(402, 12):  no member named 'Visible' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(403, 12):  no member named 'OnClick' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(404, 12):  no member named 'OnContextPopup' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(405, 12):  no member named 'OnDblClick' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(406, 12):  no member named 'OnDragDrop' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(407, 12):  no member named 'OnDragOver' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(408, 12):  no member named 'OnEndDock' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(409, 12):  no member named 'OnEndDrag' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(410, 12):  no member named 'OnGesture' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(411, 12):  no member named 'OnMouseActivate' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(412, 12):  no member named 'OnMouseDown' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(413, 12):  no member named 'OnMouseEnter' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(414, 12):  no member named 'OnMouseLeave' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(415, 12):  no member named 'OnMouseMove' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(416, 12):  no member named 'OnMouseUp' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(417, 12):  no member named 'OnResize' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(418, 12):  no member named 'OnStartDock' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(419, 12):  no member named 'OnStartDrag' in 'Skia::Vcl::TSkCustomWinControl'
[C++ Error] Skia.Vcl.hpp(421, 98):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(421, 108):  member initializer 'TCustomControl' does not name a non-static data member or base class
[C++ Error] Skia.Vcl.hpp(446, 6):  'Extctrls' is not a class, namespace, or enumeration
[C++ Error] Vcl.ExtCtrls.hpp(43, 10):  'Extctrls' declared here
[C++ Error] Skia.Vcl.hpp(929, 7):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(934, 18):  'Controls' is not a class, namespace, or enumeration
[C++ Error] Vcl.Forms.hpp(71, 10):  'Controls' declared here
[C++ Error] Skia.Vcl.hpp(1244, 12):  no member named 'AutoSize' in 'Skia::Vcl::TSkLabel'
[C++ Error] Skia.Vcl.hpp(1245, 12):  no member named 'BiDiMode' in 'Skia::Vcl::TSkLabel'
[C++ Error] Skia.Vcl.hpp(1247, 12):  no member named 'ParentBiDiMode' in 'Skia::Vcl::TSkLabel'
[C++ Error] Skia.Vcl.hpp(1259, 8):  no member named 'GetInterface' in 'Skia::Vcl::TSkLabel'

**Comment by viniciusfbb** [comment, marked as answer]
Hi @saimmm! There are another issue
With the same subject: #107

**Reply by saimmm** [reply]
thanks
